cmd/mdatagen/internal/templates/feature_gates.go.tmpl

```diff
@@ -0,0 +1,26 @@
+// Code generated by mdatagen. DO NOT EDIT.
+
+package {{ .Package }}
+
+import (
+	"go.opentelemetry.io/collector/featuregate"
+)
+
+func init() {
+    for _, fg := range {{ .FeatureGates }} {
+    	var opts []featuregate.RegisterOption
+        if fg.FromVersion != "" {
+            opts = append(opts, featuregate.WithRegisterFromVersion(fg.FromVersion))
+        }
+        if fg.Description != "" {
+            opts = append(opts, featuregate.WithRegisterDescription(fg.Description))
+        }
+        if fg.ReferenceURL != "" {
+            opts = append(opts, featuregate.WithRegisterReferenceURL(fg.ReferenceURL))
+        }
+        if fg.ToVersion != "" {
+                opts = append(opts, featuregate.WithRegisterToVersion(fg.ToVersion ))
+        }
+        _ = featuregate.GlobalRegistry().MustRegister(fg.ID, fg.Stage, opts...)
```

Comment:
We need to assign a name here if the feature gate is not stable, since we need to use it in code. The name can be based on the ID in some way (this may be the hardest part of the template so if you want we can leave that for last)